GoogleForCreators/web-stories-wp
# Popups in sidebar and floating menu repositioning

## Context


Now that we are moving the panels to one side, custom use cases of Popup need their placements updated. 

## Summary

Double checked all use cases of tooltip and pop throughout the editor. Updated Color Picker popup , Publish time popup, Color swatch tooltip, Eye dropper tooltip, Alignment tooltip, Style Presets popup.

This pr is branched off `sidebar`

## Relevant Technical Choices
- `ColorInput` was using `dock` ref for placement. Made this dynamic, since we are reusing color in the floating menu. 
- added `dock` to `Tooltip` so we can place the first alignment tooltip at the left most of the screen (in RTL mode, this is set to bottom end since dock would force it in the center of the canvas)
- added yet another flag to `Popup` this one for resetting the x -offset. In the case of RTL, "pure" uses of `Popup`, can render off screen eg. floating menu's color picker. This forces the popup inward.
- created a const file for spacing requirements depending on color input's use case. 



## To-do
n/a



## User-facing changes
||LTR|RTL|
|--|--|--|
|Color swatch tooltip|![color swatch](https://user-images.githubusercontent.com/1820266/158255029-564c8273-4b1b-4253-8775-88a92cb0d960.png)|![color swatch RTL](https://user-images.githubusercontent.com/1820266/158255068-7cf6b56f-6c1d-4d69-b21d-672ddf7ff6b7.png)|
|Color dropper tooltip|![color dropper](https://user-images.githubusercontent.com/1820266/158255166-2a51ba0a-0628-4055-b9aa-40b35dc7e17e.png)|![eyedropper RTL](https://user-images.githubusercontent.com/1820266/158255208-3e716bb3-46aa-4b4a-8cd3-1f804ba96713.png)|
|Color picker sidebar|![color picker](https://user-images.githubusercontent.com/1820266/158255278-bda76ea8-0d96-440f-ae42-e73c052254a0.png)|![color picker RTL](https://user-images.githubusercontent.com/1820266/158255294-2c6df1ca-96bd-4bdd-9490-93e695f82e7e.png)|
|Color picker floating menu default|![floating menu picker default](https://user-images.githubusercontent.com/1820266/158255425-e3cf793f-e000-4331-8d8f-e8f7ef418aa5.png)|![floating menu RTL](https://user-images.githubusercontent.com/1820266/158255449-2d26b30e-0f03-4b56-a4cf-ecd2e0f453ea.png)|
|Floating Menu edge case|![floating menu picker](https://user-images.githubusercontent.com/1820266/158255548-9aa555b6-a19c-4e93-8c72-df0c6e3a7b52.png)|![floating menu RTL edge](https://user-images.githubusercontent.com/1820266/158255575-4bfcb315-8ab0-4a19-9fcf-35d3c6d63326.png)|
|Alignment tooltip|![alignment](https://user-images.githubusercontent.com/1820266/158255664-357f40ef-b2b3-488a-9d54-6b0310ba5754.png)|![alignment RTL](https://user-images.githubusercontent.com/1820266/158255680-844f7bca-0999-4420-bf9c-524d7d2cccb0.png)|
|Style presets popup|![style presets](https://user-images.githubusercontent.com/1820266/158255739-fdeb58ca-33af-4a54-8556-0086676cf00c.png)|![style presets RTL](https://user-images.githubusercontent.com/1820266/158255770-6578a803-2af5-4e81-b18d-885790e2cb5b.png)|
|Publish time popup|![publish](https://user-images.githubusercontent.com/1820266/158255860-fd9c7dd6-5adb-4e8d-8f74-6cee2a2cb934.png)|![publish RTL](https://user-images.githubusercontent.com/1820266/158255895-42a4d3e4-0dc3-4ed2-8bb3-4f92cf2c765b.png)|



## Testing Instructions






This PR can be tested by following these steps:

1. Please check all tooltips and popups I may have missed. Known updates to: 
- Color swatch tooltip
- Color dropper tooltip
- Color picker sidebar
- Color picker floating menu
- Alignment tooltip
- Style presets popup
- Publish time popup  (used in publish modal and sidebar)


## Reviews

### Does this PR have a security-related impact?
no


### Does this PR change what data or activity we track or use?
no


### Does this PR have a legal-related impact?
no


## Checklist




---



Fixes #10803 
Fixes #10825
